### PR TITLE
Kasper

### PR DESCRIPTION
**Pull request by Kasper1999**
Task 1

### CODE REVIEW

File: src/components/Main/Button/Button.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+Button.propTypes = {
+  title: PropTypes.string.isRequired,
+};
+
+export default connect(
+  state => ({
+    weatherMainStore: state
```
Comment (AV29) [inline src/components/Main/Button/Button.jsx]: same here... pick a name

File: src/components/Main/Main.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  }
+  
+  render() {
+    return (
+    <div className = {styles.main} style={ this.backImg }>
+        <div>
+            <p className = {styles.city}> {this.props.city} </p>
+            <p className = {styles.temperature}> {this.props.temperature} &#8451; </p>
+        </div>
+        <Button title="More"></Button>
```
Comment (AV29) [inline src/components/Main/Main.jsx]: omit closing tag. Use self closing

File: src/index.jsx [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
-import App from './App.jsx';
 
-ReactDOM.render(<App />, document.getElementById('app'));
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+
+import App from './components/App.jsx';
+import reducer from './reducers';
+  
+  const store = createStore(reducer,  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
```
Comment (AV29) [inline src/index.jsx]: spacing

File: src/reducers/mainState.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const instate = {
```
Comment (AV29) [inline src/reducers/mainState.jsx]: initialState ?

File: src/components/Main/styles.css [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+.city, .temperature {
+    width: 40%;
+    min-width: 210px;
+    margin: 2vmin 5vmin;
+    padding: 10px 10px 10px 2vw;
+    font-size: 52pt;
+    color: rgb(36, 35, 71);
+    background-color: #44ebebbd;
```
Comment (AV29) [inline src/components/Main/styles.css]: Either stick to rgba or hex

File: src/components/Main/Button/styles.css [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+.button button {
+    padding: 10px;
+  border: none;
+  font: inherit;
+  color: inherit;
+  background-color: transparent;
+  position: absolute;
+    bottom: 5vh;
```
Comment (AV29) [inline src/components/Main/Button/styles.css]: spacing

File: src/components/App.jsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    );
+  }
+}
+
+export default connect(
+  state => ({
+    weatherMainStore: state
+  }),
+  dispatch => ({})
+)(App);
```
Comment (AV29) [inline src/components/App.jsx]: don't forget about propTypes. They should be added to each component that one way or another consumes some props.

File: src/components/Main/Button/Button.jsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import { connect } from 'react-redux'
+
+class Button extends React.PureComponent {
+  changeMain(){
+    this.props.onChangeMain(!this.props.weatherMainStore.main.mainState);
+  }
+  render() {
+    return (
+      <div className = {styles.button}>
+        <button onClick = {this.changeMain.bind(this)}> {this.props.title} </button>
```
Comment (AV29) [inline src/components/Main/Button/Button.jsx]: Is better to bind methods once. Say, in a constructor.

File: src/components/CityList/City/City.jsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    </div>
+    );
+}
+
+City.propTypes = {
+    title: PropTypes.string.isRequired,
+    // image:
+}
+
+// NavBarElement.defaultProps = {
```
Comment (ximet) [inline src/components/CityList/City/City.jsx]: Better use defaultProps in this case.

File: src/components/CityList/CityList.jsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+];
+
+class CityList extends React.Component {  
+  render() {
+    return (
+    <div className = {styles.city_weather}>
+
+      {
+        map(CityWeather, item => (
+          <City key={item.key} city={item.city} weather={item.weather} weather_img={ item.weather.substring(1) > 12 ? (item.weather.substring(1) > 17 ?  weatherImg1 : weatherImg2) : weatherImg3 }></City>
```
[the remark this answers is not in the document]
Comment (ximet) [reply src/components/CityList/CityList.jsx]: Better:
```
<City key={item.key} 
          city={item.city} 
          weather={item.weather} 
          weather_img={ img } />
```

File: src/components/NavBar/NavBar.jsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import NavBarElement from './NavBarElement/NavBarElement.jsx';
+import { map } from 'lodash';
```
Comment (ximet) [inline src/components/NavBar/NavBar.jsx]: You can avoid lodash in this case

**Pull request closed without merging**